### DISCUSSION TITLE
$unsafe helper

### DISCUSSION THREAD

**Discussion by SimoTod** [alpine-collective/alpine-magic-helpers#98] [Ideas] [answered]
To be used inside a x-html directory. It would roughly work in this way -> https://codepen.io/SimoTod/pen/zYoMayv

Known limitations:
- it would run the script tags even if you use the helper in other directives, the helper doesn't know which directive invoked it.
- the added scripts are appended to the bottom of the alpine component (my example above uses body just because it was fast to prove), so they won't be removed if x-html changes

**Comment by KevinBatdorf** [comment, marked as answer]
Could this some how work to load in something like this:

```html
// external-file.js
<div x-data="myComponent"></div>
<script>
window.myComponent = function() {}
</script>
```

then 
```html
<div x-data x-html="content.length ? $unsafe(content) : 'loading...'"></div>
// content would be fetched and be the contents of external-file.js
```

Essentially making for an easy way to include partials?

**Reply by SimoTod** [reply]
It should work but people needs to be careful with naming conflicts

**Comment by SimoTod** [comment]
Maybe we should call it in a different way? $eval?

**Reply by KevinBatdorf** [reply]
`$unsafeHTML()` maybe? 

We could probably use `$eval()` as a wrapper for `saferEval()`

**Reply by SimoTod** [reply]
It feels weird to call it "unsafe" but I guess it's what it does.

**Reply by KevinBatdorf** [reply]
I've been learning Rust on the side and they use "unsafe Rust" there often so to me it feels right.
